src/pages/documentation/building-mobile-extensions.md

```diff
@@ -8,6 +8,25 @@ To support customer-specific features, and allow for the greatest flexibility, A
 
 Extensions allow customers to extend the Experience Platform SDKs with their own code. This includes listening for and dispatching events, reading the shared state of any registered extension, and sharing the state of the current extension. The application can use the extension to monitor for information that Adobe does not expose by default. It can also use the extension to modify Experience Platform SDK internal operations. For example, the application can use the extension to add additional data to messages that are sent or by sending data to other systems.
 
+## Prerequisites to building an extension
+
+Before you build an extension, complete the following tasks:
+
+### Ensure that you are using the Adobe Experience Platform SDKs
+
+  Extensions extend the behavior of these SDKs.
+
+### Ensure that you can accomplish your goals only by using an extension
+
+  To determine your goals, think about the following questions:
+
+  * Do you need access to data that is not already exposed via the Adobe Experience Platform SDKs?    
+  * Do you need to be notified when messages will be sent, or data is being collected by the Adobe Experience Platform SDKs?   
+  * Do you need to add data to or modify data for outgoing messages?    
+  * Do you need to expose data to other extensions or to rules processing?
+
+  If your answer to any of these questions is **yes**, build the extension.
+
 ## Namespace conventions
```

Comment:
nit: can we use h3 here as a prerequisite for building extensions and rename to "Review the namespace conventions"